indexing-solr/src/main/java/edu/unc/lib/boxc/indexing/solr/filter/SetDescriptiveMetadataFilter.java

```diff
@@ -151,43 +153,49 @@ private void extractTitles(Element mods, IndexDocumentBean idb) {
         }
     }
 
-    private void extractNamesAndAffiliations(Element mods, IndexDocumentBean idb, boolean splitDepartments) {
+    private void extractNames(Element mods, IndexDocumentBean idb) {
         List<Element> names = mods.getChildren("name", JDOMNamespaceUtil.MODS_V3_NS);
         List<String> creators = new ArrayList<>();
         List<String> contributors = new ArrayList<>();
+        List<String> creatorsContributors = new ArrayList<>();
 
         for (Element nameEl : names) {
             // First see if there is a display form
             String nameValue = formatName(nameEl);
 
             if (nameValue != null) {
-                contributors.add(nameValue);
-
                 List<Element> roles = nameEl.getChildren("role", JDOMNamespaceUtil.MODS_V3_NS);
-                // Person is automatically a creator if no role is provided.
-                boolean isCreator = roles.size() == 0;
-                if (!isCreator) {
+                // Person is automatically a contributor if no role is provided.
+                boolean isContributor = roles.size() == 0;
+                boolean isCreator = false;
+                List<String> creatorList = Arrays.asList("creator", "author", "interviewer", "interviewee");
+                if (!isContributor) {
                     // If roles were provided, then check to see if any of them are creators.  If so, store as creator.
-                    for (Element role: roles) {
+                    for (Element role : roles) {
                         List<Element> roleTerms = role.getChildren("roleTerm", JDOMNamespaceUtil.MODS_V3_NS);
-                        for (Element roleTerm: roleTerms) {
-                            if ("creator".equalsIgnoreCase(roleTerm.getValue())) {
-                                isCreator = true;
-                                break;
-                            }
-                            if ("author".equalsIgnoreCase(roleTerm.getValue())) {
-                                isCreator = true;
-                                break;
-                            }
-                            if (isCreator) {
-                                break;
+                        if (roleTerms.isEmpty()) {
+                            isContributor = true;
+                        }  else {
+                            for (Element roleTerm : roleTerms) {
+                                String roleType = roleTerm.getValue();
```

Comment:
```suggestion
                                String roleType = roleTerm.getTextTrim();
```
Sounds like `getValue()` does a lot more stuff than we need, plus this will trim the value.